Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -45,9 +58,23 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	if (ensureMsgf(LocalPC, TEXT("ASSERT: [%i] %hs:\n'LocalPC' is null!"), __LINE__, __FUNCTION__))
 	{
 		LocalPC->OnAnyCinematicStarted.AddDynamic(this, &APSStarActor::OnAnyCinematicStarted);
+		Character->OnPlayerTypeChanged.AddUniqueDynamic(this, &ThisClass::OnPlayerTypeChanged);
 	}
 }
 
+
+// Called when the current  player was changed
+void APSStarActor::OnPlayerTypeChanged(FPlayerTag PlayerTag)
+{
+	InitStarMenuAnimation();
```

Comment:
I don't think you need OnPlayerTypeChanged delegate at all. It's already enough GameState delegate as we want to always play animation while in menu state. Please remove OnPlayerTypeChanged as redundant and test

Reply:
shall I keep as for now, since the animation isn't playing when I do character switch without playerTypeChanged? Also, not sure if it will stay alive after going back and forth with pool manager?

Reply:
@h4rdmol, I just tested, OnPlayerTypeChanged is safe to remove, animations keep working after switching the player. It does not play after switch because of next mistake: [link](https://github.com/h4rdmol/ProgressionSystem/pull/14/files#r1766029026)

Please remove OnPlayerTypeChanged as redundant